.github/workflows/npmpublish.yml

```diff
@@ -0,0 +1,34 @@
+name: "NPM Publish and GitHub tag"
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - run: yarn install
+      - run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+      # Set PACKAGE_VERSION, pull from package.json
+      - name: package-version
+        run: node -p -e '`PACKAGE_VERSION=${require("./package.json").version}`' >> $GITHUB_ENV
+      # Create the tag using the new PACKAGE_VERSION value
+      - name: Create tag
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ github.token }}
+          script: |
+            github.git.createRef({
```

Comment:
It is OK to have multiple tags with the same `PACKAGE_VERSION`, or do you only want to add tags when package.json changes?

Reply:
This is a great point.  Currently, by convention, we only pull into main on release of a new version (develop is our working branch).  But it would be nice to have additional safeguards.  Tagging is really the stretch objective here.  Before this change, there was no way to know that[ what the developer was publishing matches what was in main](https://sagebionetworks.jira.com/browse/PORTALS-792)!